substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/SecurityServicesFeature.java

```diff
@@ -659,7 +659,10 @@ private static boolean isValid(Service s) {
     private static Function<String, Class<?>> getConstructorParameterClassAccessor(ImageClassLoader loader) {
         Map<String, /* EngineDescription */ Object> knownEngines = ReflectionUtil.readStaticField(Provider.class, "knownEngines");
         Class<?> clazz = loader.findClassOrFail("java.security.Provider$EngineDescription");
-        Field consParamClassNameField = ReflectionUtil.lookupField(clazz, "constructorParameterClassName");
+        int feature = Runtime.version().feature();
+        int update = Runtime.version().update();
+        Field consParamClassField = ReflectionUtil.lookupField(clazz,
+                feature == 22 && update >= 2 ? "constructorParameterClass" : "constructorParameterClassName");
```

Comment:
I'm worried that this might not give the correct results for some EA builds. Could we test for `constructorParameterClass` field availability instead and use it if there, otherwise use `constructorParameterClassName` field?

Reply:
Done